### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by ghost** [B0ney/xmodits#25] [General]
Just a quick message to say you’re an absolute legend for this, been badly needing an application that can bulk export samples.

Thanks so much!

**Comment by jumbledFox** [comment]
Agreed! This is super useful, added bonus of being written in rust ;)

**Comment by B0ney** [comment]
> Just a quick message to say you’re an absolute legend for this, been badly needing an application that can bulk export samples.
> 
> Thanks so much!

> Agreed! This is super useful, added bonus of being written in rust ;)

Thanks! I'm glad both of you found my program useful! ^_^

**Comment by chris-roerig** [comment]
I started writing my own but quickly hit xm header issues - got frustrated and found your app. I'm amazed at how quickly this thing works! I was able to rip ~25 xm in under 1 second. Thank you

**Comment by B0ney** [comment]
> I started writing my own but quickly hit xm header issues - got frustrated and found your app. I'm amazed at how quickly this thing works! I was able to rip ~25 xm in under 1 second. Thank you

Gosh, I share your experience, dealing with the xm format was an absolute nightmare!

I'm glad you found it useful, and _very_ happy to see it work quickly for you. When I added multi-threading in v0.10.0, I was able to extract ~2500 modules in ~12 seconds, compared to the usual 45. It's pretty incredible 😁
